src/main/java/reposense/model/IgnoreCommitsList.java

```diff
@@ -0,0 +1,54 @@
+package reposense.model;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class IgnoreCommitsList {
+    private static final String COMMIT_HASH_REGEX = "^[0-9a-f]+$";
+    private static final String INVALID_COMMIT_HASH_MESSAGE =
+            "The provided commit hash, %s, contains illegal characters.";
+
+    private List<String> ignoreCommits = new ArrayList<>();
+
+    public List<String> getIgnoreCommitsList() {
+        return ignoreCommits;
+    }
+
+    public void setIgnoreCommitsList(List<String> formats) {
+        validateIgnoreCommits(formats);
```

Comment:
This is doing double job. Although the effect is marginal, I would prefer to keep a clear separation between validation and set so that future developers will not attempt to set any of the values before all fields are validated in the `update` method.